test/template-tests/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1

```diff
@@ -0,0 +1,34 @@
+﻿param(
+[Parameter(Mandatory=$true,Position=0)]
+[PSObject]
+$MainTemplateObject
+)
+
+$allApiVersions = $MainTemplateObject | 
+    Find-AzureRMTemplate -Key apiVersion -Value * -Like
+
+foreach ($av in $allApiVersions) {
+    if ($av.ApiVersion -isnot [string]) {

```

Comment:
not sure we need this since deployment will fail if this is not true

Reply:
I don't mind testing for things that will also fail a deployment if it makes a cleaner test case.

Reply:
In this case the failure is obvious enough that I don't think we need to test for it here... (i.e. remove this one) ...  but we can take these case by case